File: typescript/src/shapes/square.ts
function square(sideLen: number): Shape {
    let _sideLen: number = sideLen

    function getSideLen(): number {
        return _sideLen
    }
    return {
        computeArea: function (): number {
            return getSideLen() * getSideLen()
        }
    }
}

export { square }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    private double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }
    
    public double getArea() {
        return getSideLen() * getSideLen();
    }

    private double getSideLen() {
        return this.sideLen;
    }
}
